frame/contracts/src/migration.rs

```diff
@@ -191,10 +191,14 @@ pub trait MigrateSequence: private::Sealed {
 }
 
 /// Performs all necessary migrations based on `StorageVersion`.
-pub struct Migration<T: Config>(PhantomData<T>);
+///
+/// If `RUN_ALL_STEPS == true` and `try-runtime` is enabled, this will run all the migrations inside
+/// `on_runtime_upgrade`. This should be set to false in tests that want to ensure the step by step
+/// migration works.
+pub struct Migration<T: Config, const RUN_ALL_STEPS: bool = true>(PhantomData<T>);
```

Comment:
question: do we also run the tests in CI without try_runtime? in which case I think it would be fine to just disable the test for try_runtime and not add this extra RUN_ALL_STEPS

If not, maybe we change the name to TRY_ALL_STEPS or TEST_ALL_STEPS to make it clear that this is just intended for testing purpose.

Reply:
With the changes of the pr, we run all tests with `try-runtime`.